src/lib/scheduler/worker.cpp

```diff
@@ -44,12 +45,20 @@ void Worker::operator()() {
 
   _set_affinity();
 
+  std::mutex lock;
+  std::unique_lock<std::mutex> unique_lock(lock);
+  bool queues_are_empty;
```

Comment:
Why queue*s*? I only see a single one.

Reply:
Because, I am also checking the queues not located on the current node (work stealing).